src/schoology-extractor/schoology_extractor/api/request_client.py

```diff
@@ -294,10 +343,8 @@ def get_roles(self, page_size: int = DEFAULT_PAGE_SIZE) -> PaginatedResult:
         url = f"roles?{self._build_query_params_for_first_page(page_size)}"
         response = self.get(url)
 
-        return PaginatedResult(
-            self, page_size, response, "role", self.base_url + url
-        )
+        return PaginatedResult(self, page_size, response, "role", self.base_url + url)
 
     def _build_query_params_for_first_page(self, page_size: int):
-        assert isinstance(page_size, int)
+        assert isinstance(page_size, int), "Argument `page_size` should be of type `int`."
         return f"start=0&limit={page_size}"
```

Comment:
I think this file is starting to get too big for a single file, if it keeps growing, sooner than later it will start getting difficult to read/understand.

Reply:
Many of the lines are just from docstrings 😆 .

The paging code can be moved to a helper module.  Perhaps `get_assignments_by_section_ids` should be changed to move the looping over sections out to the `schoology_request_service` (or whatever name it gets) so that the `request_client` module is more low-level, without that business logic. There's an opportunity to refactor this in the separate ticket I created for refactoring the paging support.